src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/WhyCommand/WhyCommandRunner.cs

```diff
@@ -46,57 +46,104 @@ public static int ExecuteCommand(WhyCommandArgs whyCommandArgs)
+            foreach ((string assetsFilePath, string? projectPath) in assetsFiles)
             {
-                Project project = MSBuildAPIUtility.GetProject(projectPath);
-
-                string usingNetSdk = project.GetPropertyValue("UsingMicrosoftNETSdk");
+                LockFile? assetsFile = GetProjectAssetsFile(assetsFilePath, projectPath, whyCommandArgs.Logger);
 
-                if (!string.IsNullOrEmpty(usingNetSdk))
+                if (assetsFile != null)
                 {
-                    LockFile? assetsFile = GetProjectAssetsFile(project, whyCommandArgs.Logger);
+                    ValidateFrameworksOptionsExistInAssetsFile(assetsFile, whyCommandArgs.Frameworks, whyCommandArgs.Logger);
+
+                    Dictionary<string, List<DependencyNode>?>? dependencyGraphPerFramework = DependencyGraphFinder.GetAllDependencyGraphsForTarget(
+                        assetsFile,
+                        whyCommandArgs.Package,
+                        whyCommandArgs.Frameworks);
 
-                    if (assetsFile != null)
+                    if (dependencyGraphPerFramework != null)
                     {
-                        ValidateFrameworksOptionsExistInAssetsFile(assetsFile, whyCommandArgs.Frameworks, whyCommandArgs.Logger);
+                        whyCommandArgs.Logger.LogMinimal(
+                            string.Format(
+                                Strings.WhyCommand_Message_DependencyGraphsFoundInProject,
+                                assetsFile.PackageSpec.Name,
+                                targetPackage));
 
-                        Dictionary<string, List<DependencyNode>?>? dependencyGraphPerFramework = DependencyGraphFinder.GetAllDependencyGraphsForTarget(
-                            assetsFile,
-                            whyCommandArgs.Package,
-                            whyCommandArgs.Frameworks);
+                        DependencyGraphPrinter.PrintAllDependencyGraphs(dependencyGraphPerFramework, targetPackage, whyCommandArgs.Logger);
+                    }
+                    else
+                    {
+                        whyCommandArgs.Logger.LogMinimal(
+                            string.Format(
+                                Strings.WhyCommand_Message_NoDependencyGraphsFoundInProject,
+                                assetsFile.PackageSpec.Name,
+                                targetPackage));
+                    }
+                }
+                else
+                {
+                    anyErrors = true;
+                }
+            }
 
-                        if (dependencyGraphPerFramework != null)
+            return anyErrors ? ExitCodes.Error : ExitCodes.Success;
+        }
+
+        private static IEnumerable<(string assetsFilepath, string? projectPath)> FindAssetsFiles(string path, ILoggerWithColor logger)
+        {
+            var extension = Path.GetExtension(path);
+            if (XPlatUtility.IsJsonFile(extension))
```

Comment:
`XPlatUtility.IsJsonFile()` utility method does invoke `Path.GetExtension()` to check if the file is a json file or not. I think we can directly pass the path to the utility method.

```suggestion
            if (XPlatUtility.IsJsonFile(path))
```